### PR TITLE
Student Portal Basic Structure

### PR DESCRIPTION
**Pull request by HMayaz001**
**I transformed my application from javascript to typescript which required following changes:**

- > Created a new project with the command   _**npx create-react-app my-app --template typescript_**
- > Replaced all the typescript files with the javascript 
- > Created a new PR for my updated project

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (ahmadfaraz629) [inline public/index.html:27]: Change App Title to match you icon

File: public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+      "src": "logo512.png",
+      "type": "image/png",
+      "sizes": "512x512"
+    }
+  ],
+  "start_url": ".",
+  "display": "standalone",
+  "theme_color": "#000000",
+  "background_color": "#ffffff"
```
Comment (ahmadfaraz629) [inline public/manifest.json:24]: change short name and name according to your app

File: src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import "./App.css";
+const initialState = (window as any).initialReduxState;
+// const store = configureStore(initialState);
+
+function App() {
+  return (
+    <></>
+    // <Provider store={store}>
+    //   <StudentPortal />
+    // </Provider>
```
Comment (ahmadfaraz629) [inline src/App.tsx:12]: Your structure should contain basic implementation of of redux with a component.

redux itself can be empty and component can return <></> for now.

File: src/state/configureStore.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+// export default function configureStore(
+//  initialState: IApplicationState
+// ): Store<IApplicationState> {
+//  const middlewares = applyMiddleware();
+//  // Create Store
+//  const store = createStore(rootReducer, initialState, middlewares);
+//  return store;
+// }
+
```
Comment (ahmadfaraz629) [inline src/state/configureStore.ts:12]: Store connection should be complete with all middleware

File: src/state/configureStore.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+export{}
```
Comment (ahmadfaraz629) [inline src/state/configureStore.ts:13]: new line(EOF) missing

File: src/state/ducks/index.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+// 	post: IPostState;
+// }
+// export interface IMetaAction extends MetaAction<TypeConstant, IMeta> {}
+// export interface IReducerAction<TPayload>
+// 	extends Action<TypeConstant>,
+// 		PayloadAction<TypeConstant, TPayload> {}
+// export const rootReducer = combineReducers<IApplicationState>({
+// 	post: postReducer
+// });
+
```
Comment (ahmadfaraz629) [inline src/state/ducks/index.ts:13]: Create an empty student reducer slice and add it in root reducer.

File: src/state/ducks/index.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+export{ }
```
Comment (ahmadfaraz629) [inline src/state/ducks/index.ts:14]: New line EOF missing

File: src/state/index.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+
+// src/state/index.ts
+
+// import configureStore from "./configureStore.dev";
+// export default configureStore;
+
```
Comment (ahmadfaraz629) [inline src/state/index.ts:6]: Store connection should be complete here as well.

**Comment by nofalfayyaz** [comment]
Good work @HMayaz001. Let's use only small letters for commit messages. Mix of initial capital and initial small letter will create inconsistency
Screenshot 2022-06-17 at 11 58 42 AM

**Comment by HMayaz001** [comment]
Noted.

File: src/state/configureStore.ts (hunk 5 above)
Comment (HMayaz001) [reply src/state/configureStore.ts]: Done

**Review by ahmadfaraz629 — APPROVED** [review]
LGTM 👍

**Pull request merged**